sympy/combinatorics/schur_number.py

```diff
@@ -0,0 +1,161 @@
+"""
+The Schur number S(k) is the largest integer n for which the interval [1,n]
+can be partitioned into k sum-free sets.(http://mathworld.wolfram.com/SchurNumber.html)
+"""
+import math
+from sympy.core import S
+from sympy.core.basic import Basic
+from sympy.core.function import Function
+from sympy.core.numbers import Integer
+
+
+class SchurNumber(Function):
+    """
+    This function creates a SchurNumber object
+    which is evaluated for k <= 4 otherwise only
+    the lower bound information can be retrieved.
+
+    Examples
+    ========
+
+    >>> from sympy.combinatorics.schur_number import SchurNumber
+
+    Since S(3) = 13, hence the output is a number
+    >>> SchurNumber(3)
+    13
+
+    We don't know the schur number for values greater than 4, hence
+    only the object is returned
+    >>> SchurNumber(6)
+    SchurNumber(6)
+
+    Now, the lower bound information can be retrieved using lower_bound()
+    method
+    >>> SchurNumber(6).lower_bound()
+    364
+
+    """
+
+    @classmethod
+    def eval(cls, k):
+        if k.is_Number:
+            if k is S.Infinity:
+                return S.Infinity
+            if k.is_zero:
+                return 0
+            if not k.is_integer or k.is_negative:
+                raise ValueError("k should be a positive integer")
+            first_known_schur_numbers = {1: 1, 2: 4, 3: 13, 4: 44}
+            if k <= 4:
+                return Integer(first_known_schur_numbers[k])
+
+    def lower_bound(self):
+        f_ = self.args[0]
+        return (3**f_ - 1)/2
+
+
+class SchurSubsetsNumber(Function):
```

Comment:
I wonder what is schur subsets number?
And why is this needed symbolically if it is simply `log(2n+1, 3)`

Reply:
Well ```SchurSubsetsNumber``` gives us the upper bound of the number of sum free subsets ```k``` given the value ```n``` i.e. given the Schur Number ```S(k)``` what is the upper bound of k. I am little unsure whether it is needed symbolically or not, but the author of the previous PR #14493 used a function to return the value for computations required in ```schur_partition``` function.

Reply:
Unfortunately, I don't see any symbolic use of computations only for giving upper bounds. I'd expect it to be polished like `SchurNumber` or you should rather make this as a private function.

I'm thinking of an idea of having a function like `IntegersSumFreeSubsets` that can evaluate for known values, like how `SchurNumber` does now. And grouped with utility functions.

I think that the documentation is poor without a reference where `log(2n+1, 3)` appears. Though you may have to ask the original author or find the sources where it appears.

Reply:
> Unfortunately, I don't see any symbolic use of computations only for giving upper bounds.

Agreed.

Reply:
@sylee957 I would make it as a private function, but by private function you mean just using the formula and evaluating it right?

Also, I would ask the original author about the ```log(2n+1, 3)``` soon.

Reply:
For private function, you should prefix stuff with `_`